src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1756,16 +1756,32 @@ private void visitTypeAlias(JSType registryType, TypedVar symbol) {
 
     private void visitTypeAlias(
         JSType registryType, String unqualifiedName, boolean emitNeverBrand) {
-      emit("type");
-      emit(unqualifiedName);
-      emit("=");
-      visitType(registryType, true, false);
+      visitTypeAliasHelper(registryType, unqualifiedName);
       // emit a brand to prevent accidental compatibility of values with an enum.
       if (emitNeverBrand) emit("&{clutzEnumBrand: never}");
       emit(";");
       emitBreak();
     }
 
+    private void visitTypeAlias(
+        JSType registryType, String unqualifiedName, Set<String> literalTypes) {
+      visitTypeAliasHelper(registryType, unqualifiedName);
+      emit("&{clutzEnumBrand: never}");
+      // Inline all literal types in the type alias.
+      for (String n : literalTypes) {
+        emit("|'" + n + "'");
+      }
+      emit(";");
+      emitBreak();
+    }
+
+    private void visitTypeAliasHelper(JSType registryType, String unqualifiedName) {
```

Comment:
Helper is not a very descriptive name. Maybe 'emitTypeAliasPrefix'?

Reply:
Done.